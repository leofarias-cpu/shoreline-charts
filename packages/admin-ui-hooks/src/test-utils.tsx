import 'jest-location-mock'

export function jestWindowMock() {
  let history: string[] = [window.location.pathname]
  let index = 0
  let popstateCallback = () => {}

  Object.defineProperties(window.history, {
    pushState: {
      value: (data: { path: string }) => {
        history = [...history.splice(0, index + 1), data.path]
        index += 1

        window.location.assign(data.path)
        popstateCallback()
      },
    },
    back: {
      value: () => {
        if (index <= 0) return

        index -= 1
        window.location.assign(history[index])
        popstateCallback()
      },
    },
    forward: {
      value: () => {
        if (index >= history.length) return

        index += 1
        window.location.assign(history[index])
        popstateCallback()
      },
    },
  })
  Object.defineProperties(window, {
    addEventListener: {
      value: (name: string, fn: () => void) => {
        if (name === 'popstate') {
          popstateCallback = fn
        }
      },
    },
  })
}

export function setQueryParam(query: Record<string, string | undefined>) {
  const queryParams = new URLSearchParams(window.location.search)

  Object.entries(query).forEach((element: [string, string | undefined]) => {
    if (!element[1]) {
      queryParams.delete(element[0])

      return
    }

    queryParams.set(element[0], element[1])
  })

  window.location.assign(`?${queryParams.toString()}`)
}

export function getCurrentQuery() {
  const queryParams = new URLSearchParams(window.location.search)

  return Object.fromEntries(queryParams.entries())
}

export function cleanQueryParam() {
  window.location.assign('')
}
