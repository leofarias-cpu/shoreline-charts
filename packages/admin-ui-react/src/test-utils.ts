import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { ThemeProvider } from './system'

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
