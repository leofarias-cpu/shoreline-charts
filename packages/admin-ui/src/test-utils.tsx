import React, { useRef } from 'react'
import { axe } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { ThemeProvider } from '@vtex/admin-core'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export * from '@testing-library/react'
export { render, withState, axe, jestMatchMedia, renderHook, act }
