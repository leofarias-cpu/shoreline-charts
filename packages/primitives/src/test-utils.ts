import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { ThemeProvider } from '@vtex/admin-core'
import type { ReactElement } from 'react'

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export * from '@testing-library/react'
export { render }
