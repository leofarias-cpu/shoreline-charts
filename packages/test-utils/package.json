{
  "name": "@vtex/shoreline-test-utils",
  "version": "0.0.0",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "engines": {
    "node": ">=12"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "tsup": "7.2.0",
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.3",
    "@testing-library/jest-dom": "6.1.4",
    "@testing-library/react": "14.1.2"
  }
}
