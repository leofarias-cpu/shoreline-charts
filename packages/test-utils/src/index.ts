export * from './react'
export * from './vitest'
export * from './expect'
