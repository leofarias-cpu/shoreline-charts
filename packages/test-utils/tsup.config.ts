import { defineConfig } from 'tsup'

export default defineConfig({
  entry: ['src/index.ts'],
  format: ['esm'],
  external: ['vitest'],
  sourcemap: true,
  clean: true,
  dts: true,
})
