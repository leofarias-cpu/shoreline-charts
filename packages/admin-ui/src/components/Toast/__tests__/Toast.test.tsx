import React from 'react'
import { render, axe } from '../../../test-utils'

import { Toast, ToastProvider } from '../index'

const toastProps = {
  onClear: () => {},
  id: 'toast-id',
  dedupeKey: 'toast-id',
  message: 'This is a toast!',
  action: {
    label: 'Action',
    onClick: () => {},
  },
  shouldRemove: true,
  dismissible: true,
}

describe('Toast', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <ToastProvider>
        <Toast {...toastProps} />
        <Toast {...toastProps} type="success" />
        <Toast {...toastProps} type="warning" />
        <Toast {...toastProps} type="error" />
      </ToastProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should not have a11y violations', async () => {
    const { container } = render(
      <ToastProvider>
        <Toast {...toastProps} />
      </ToastProvider>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
