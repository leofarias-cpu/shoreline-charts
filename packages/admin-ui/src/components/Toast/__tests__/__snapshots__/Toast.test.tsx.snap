// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should match snapshot 1`] = `
<DocumentFragment>
  .emotion-11 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-width: 16.125rem;
  width: auto;
  min-height: 4.5rem;
  height: auto;
  max-height: 4.5rem;
  padding: 1rem;
  box-shadow: 0rem 0rem 1.5rem 0rem rgba(0,0,0,0.05);
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  border-color: #E0E2E7;
  pointer-events: all;
  overflow: hidden;
  padding-top: 1rem;
  padding-bottom: 1rem;
  color: #323845;
  background-color: #FFFFFF;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 >*:first-child {
  margin-right: 0.75rem;
}

.emotion-0 {
  min-width: 24px;
  min-height: 24px;
  width: 24px;
  height: 24px;
  color: #2953B2;
}

.emotion-1 {
  text-align: start;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  line-height: 1.25;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  font-size: 0.875rem;
  height: 40px;
  width: auto;
  padding-left: 1rem;
  padding-right: 1rem;
  color: #2953B2;
  background-color: transparent;
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-5:hover {
  background-color: rgba(50,56,69,0.04);
}

.emotion-5:active {
  background-color: rgba(50,56,69,0.08);
}

.emotion-5:disabled {
  color: #898F9E;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  font-size: 0.75rem;
  height: 32px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  color: currentColor;
  background-color: transparent;
}

.emotion-9:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-9:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-9:hover {
  background-color: rgba(50,56,69,0.04);
}

.emotion-9:active {
  background-color: rgba(50,56,69,0.08);
}

.emotion-9:disabled {
  color: #898F9E;
}

.emotion-9 svg {
  width: 20px;
  height: 20px;
  min-width: 20px;
  min-height: 20px;
  margin: 0.25rem;
}

.emotion-6 {
  min-width: 24px;
  min-height: 24px;
  width: 24px;
  height: 24px;
}

<div
    class="emotion-11"
    role="alert"
  >
    <div
      class="emotion-2"
    >
      <svg
        class="emotion-0"
        fill="none"
        viewBox="0 0 24 24"
      >
        <path
          d="M12.905 5.65676H12.907M12.907 5.65676H11.094C8.592 5.65676 6.563 7.68476 6.563 10.1878V12.9848C6.563 13.5148 6.352 14.0238 5.977 14.3988L5.336 15.0398C4.961 15.4148 4.75 15.9238 4.75 16.4538C4.75 17.4978 5.596 18.3438 6.64 18.3438H17.361C18.405 18.3438 19.251 17.4978 19.251 16.4538C19.251 15.9238 19.04 15.4148 18.665 15.0398L18.024 14.3988C17.649 14.0238 17.438 13.5148 17.438 12.9848V10.1881C17.438 7.68506 15.409 5.65676 12.907 5.65676Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M9.70801 18.3438V18.7088C9.70801 19.9738 10.734 20.9998 12 20.9998V20.9998C13.266 20.9998 14.292 19.9738 14.292 18.7078V18.3428"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M13.833 5.751V4.833C13.833 3.821 13.013 3 12 3V3C10.987 3 10.167 3.821 10.167 4.833V5.751"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
      <p
        class="emotion-1"
      >
        This is a toast!
      </p>
    </div>
    <div
      class="emotion-10"
    >
      <button
        class="emotion-5"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
             Action
          </div>
        </div>
      </button>
      <button
        aria-label="Close toast"
        class="emotion-9"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
            <svg
              class="emotion-6"
              fill="none"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6L18 18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M18 6L6 18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
             
          </div>
        </div>
      </button>
    </div>
  </div>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 >*:first-child {
  margin-right: 0.75rem;
}

.emotion-1 {
  text-align: start;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  line-height: 1.25;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  font-size: 0.875rem;
  height: 40px;
  width: auto;
  padding-left: 1rem;
  padding-right: 1rem;
  color: #2953B2;
  background-color: transparent;
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-5:hover {
  background-color: rgba(50,56,69,0.04);
}

.emotion-5:active {
  background-color: rgba(50,56,69,0.08);
}

.emotion-5:disabled {
  color: #898F9E;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  font-size: 0.75rem;
  height: 32px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  color: currentColor;
  background-color: transparent;
}

.emotion-9:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-9:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-9:hover {
  background-color: rgba(50,56,69,0.04);
}

.emotion-9:active {
  background-color: rgba(50,56,69,0.08);
}

.emotion-9:disabled {
  color: #898F9E;
}

.emotion-9 svg {
  width: 20px;
  height: 20px;
  min-width: 20px;
  min-height: 20px;
  margin: 0.25rem;
}

.emotion-0 {
  min-width: 24px;
  min-height: 24px;
  width: 24px;
  height: 24px;
}

.emotion-11 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-width: 16.125rem;
  width: auto;
  min-height: 4.5rem;
  height: auto;
  max-height: 4.5rem;
  padding: 1rem;
  box-shadow: 0rem 0rem 1.5rem 0rem rgba(0,0,0,0.05);
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  border-color: #8FC2B1;
  pointer-events: all;
  overflow: hidden;
  padding-top: 1rem;
  padding-bottom: 1rem;
  color: #323845;
  background-color: #F0F8F5;
}

<div
    class="emotion-11"
    role="alert"
  >
    <div
      class="emotion-2"
    >
      <svg
        class="emotion-0"
        fill="none"
        viewBox="0 0 24 24"
      >
        <path
          clip-rule="evenodd"
          d="M12 21C7.029 21 3 16.971 3 12C3 7.029 7.029 3 12 3C16.971 3 21 7.029 21 12C21 16.971 16.971 21 12 21Z"
          fill="#44B172"
          fill-rule="evenodd"
        />
        <path
          d="M16 10L11 15L8 12"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
      <p
        class="emotion-1"
      >
        This is a toast!
      </p>
    </div>
    <div
      class="emotion-10"
    >
      <button
        class="emotion-5"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
             Action
          </div>
        </div>
      </button>
      <button
        aria-label="Close toast"
        class="emotion-9"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
            <svg
              class="emotion-0"
              fill="none"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6L18 18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M18 6L6 18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
             
          </div>
        </div>
      </button>
    </div>
  </div>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 >*:first-child {
  margin-right: 0.75rem;
}

.emotion-1 {
  text-align: start;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  line-height: 1.25;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  font-size: 0.875rem;
  height: 40px;
  width: auto;
  padding-left: 1rem;
  padding-right: 1rem;
  color: #2953B2;
  background-color: transparent;
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-5:hover {
  background-color: rgba(50,56,69,0.04);
}

.emotion-5:active {
  background-color: rgba(50,56,69,0.08);
}

.emotion-5:disabled {
  color: #898F9E;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  font-size: 0.75rem;
  height: 32px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  color: currentColor;
  background-color: transparent;
}

.emotion-9:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-9:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-9:hover {
  background-color: rgba(50,56,69,0.04);
}

.emotion-9:active {
  background-color: rgba(50,56,69,0.08);
}

.emotion-9:disabled {
  color: #898F9E;
}

.emotion-9 svg {
  width: 20px;
  height: 20px;
  min-width: 20px;
  min-height: 20px;
  margin: 0.25rem;
}

.emotion-0 {
  min-width: 24px;
  min-height: 24px;
  width: 24px;
  height: 24px;
}

.emotion-11 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-width: 16.125rem;
  width: auto;
  min-height: 4.5rem;
  height: auto;
  max-height: 4.5rem;
  padding: 1rem;
  box-shadow: 0rem 0rem 1.5rem 0rem rgba(0,0,0,0.05);
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  border-color: #E5C38E;
  pointer-events: all;
  overflow: hidden;
  padding-top: 1rem;
  padding-bottom: 1rem;
  color: #323845;
  background-color: #FFF9EE;
}

<div
    class="emotion-11"
    role="alert"
  >
    <div
      class="emotion-2"
    >
      <svg
        class="emotion-0"
        fill="none"
        viewBox="0 0 24 24"
      >
        <path
          clip-rule="evenodd"
          d="M3 14.381V9.61899C3 8.01699 3.852 6.53599 5.236 5.72999L9.736 3.11099C11.135 2.29699 12.864 2.29699 14.263 3.11099L18.763 5.72999C20.148 6.53599 21 8.01699 21 9.61899V14.381C21 15.983 20.148 17.464 18.764 18.27L14.264 20.889C12.865 21.703 11.136 21.703 9.737 20.889L5.237 18.27C3.852 17.464 3 15.983 3 14.381Z"
          fill="#FFC165"
          fill-rule="evenodd"
        />
        <path
          d="M12 12V7.5"
          stroke="#323845"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M11.999 15.5C11.861 15.5 11.749 15.612 11.75 15.75C11.75 15.888 11.862 16 12 16C12.138 16 12.25 15.888 12.25 15.75C12.25 15.612 12.138 15.5 11.999 15.5"
          stroke="#323845"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
      <p
        class="emotion-1"
      >
        This is a toast!
      </p>
    </div>
    <div
      class="emotion-10"
    >
      <button
        class="emotion-5"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
             Action
          </div>
        </div>
      </button>
      <button
        aria-label="Close toast"
        class="emotion-9"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
            <svg
              class="emotion-0"
              fill="none"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6L18 18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M18 6L6 18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
             
          </div>
        </div>
      </button>
    </div>
  </div>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 >*:first-child {
  margin-right: 0.75rem;
}

.emotion-1 {
  text-align: start;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  line-height: 1.25;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  font-size: 0.875rem;
  height: 40px;
  width: auto;
  padding-left: 1rem;
  padding-right: 1rem;
  color: #2953B2;
  background-color: transparent;
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-5:hover {
  background-color: rgba(50,56,69,0.04);
}

.emotion-5:active {
  background-color: rgba(50,56,69,0.08);
}

.emotion-5:disabled {
  color: #898F9E;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  font-size: 0.75rem;
  height: 32px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  color: currentColor;
  background-color: transparent;
}

.emotion-9:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-9:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-9:hover {
  background-color: rgba(50,56,69,0.04);
}

.emotion-9:active {
  background-color: rgba(50,56,69,0.08);
}

.emotion-9:disabled {
  color: #898F9E;
}

.emotion-9 svg {
  width: 20px;
  height: 20px;
  min-width: 20px;
  min-height: 20px;
  margin: 0.25rem;
}

.emotion-0 {
  min-width: 24px;
  min-height: 24px;
  width: 24px;
  height: 24px;
}

.emotion-11 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-width: 16.125rem;
  width: auto;
  min-height: 4.5rem;
  height: auto;
  max-height: 4.5rem;
  padding: 1rem;
  box-shadow: 0rem 0rem 1.5rem 0rem rgba(0,0,0,0.05);
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  border-color: #EDB6B6;
  pointer-events: all;
  overflow: hidden;
  padding-top: 1rem;
  padding-bottom: 1rem;
  color: #323845;
  background-color: #FFF8F8;
}

<div
    class="emotion-11"
    role="alert"
  >
    <div
      class="emotion-2"
    >
      <svg
        class="emotion-0"
        fill="none"
        viewBox="0 0 24 24"
      >
        <path
          clip-rule="evenodd"
          d="M7.414 20.414L3.586 16.586C3.211 16.211 3 15.702 3 15.172V8.828C3 8.298 3.211 7.789 3.586 7.414L7.414 3.586C7.789 3.211 8.298 3 8.828 3H15.171C15.701 3 16.21 3.211 16.585 3.586L20.413 7.414C20.789 7.789 21 8.298 21 8.828V15.171C21 15.701 20.789 16.21 20.414 16.585L16 21H8.828C8.298 21 7.789 20.789 7.414 20.414Z"
          fill="#FF6565"
          fill-rule="evenodd"
        />
        <path
          d="M9 15L15 9"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M15 15L9 9"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
      <p
        class="emotion-1"
      >
        This is a toast!
      </p>
    </div>
    <div
      class="emotion-10"
    >
      <button
        class="emotion-5"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
             Action
          </div>
        </div>
      </button>
      <button
        aria-label="Close toast"
        class="emotion-9"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
            <svg
              class="emotion-0"
              fill="none"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6L18 18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M18 6L6 18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
             
          </div>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
